api/clients/v2/coretypes/eigenda_cert.go

```diff
@@ -4,34 +4,240 @@ import (
-	NonSignerStakesAndSignature contractEigenDACertVerifier.EigenDATypesV1NonSignerStakesAndSignature
+var (
+	v3CertTypeEncodeArgs abi.Arguments
+)
+
+func init() {
+	// load the ABI and parse the dummy interface methods used to encode the cert
+	// NOTE: the only other way would be defining the certificate using go-ethereum's abi
+	// low level types which would require much boiler plate
+	certTypesBinding, err := certTypesBinding.ContractIEigenDACertTypeBindingsMetaData.GetAbi()
+	if err != nil {
+		panic(err)
+	}
+
+	v3CertTypeEncodeMethod, ok := certTypesBinding.Methods["dummyVerifyDACertV3"]
+	if !ok {
+		panic("dummyVerifyDACertV3 not found in IEigenDACertTypes ABI")
+	}
+
+	v3CertTypeEncodeArgs = v3CertTypeEncodeMethod.Inputs
+
+}
+
+// CertificateVersion denotes the version of the EigenDA certificate
+// and is interpreted from querying the EigenDACertVerifier contract's
+// CertVersion() view function
+type CertificateVersion = uint64
+
+const (
+	// starting at two since we never formally defined a V1 cert in the core code
+	VersionTwoCert   = 0x2
+	VersionThreeCert = 0x3
+)
+
+// VerifyStatusCode reflects the return byte generated by a verification call to a generic cert verifier
+type VerifyStatusCode uint8
+
+const (
+	// NULL_ERROR Unused status code. If this is returned, there is a bug in the code.
+	StatusNullError VerifyStatusCode = iota
+	// SUCCESS Verification succeeded
+	StatusSuccess
+	// INVALID_INCLUSION_PROOF Merkle inclusion proof is invalid
+	StatusInvalidInclusionProof
+	// SECURITY_ASSUMPTIONS_NOT_MET Security assumptions not met
+	StatusSecurityAssumptionsNotMet
+	// BLOB_QUORUMS_NOT_SUBSET Blob quorums not a subset of confirmed quorums
+	StatusBlobQuorumsNotSubset
+	// REQUIRED_QUORUMS_NOT_SUBSET Required quorums not a subset of blob quorums
+	StatusRequiredQuorumsNotSubset
+)
+
+// String returns a human-readable representation of the StatusCode.
+func (s VerifyStatusCode) String() string {
+	switch s {
+	case StatusNullError:
+		return "Null Error: Unused status code. If this is returned, there is a bug in the code."
+	case StatusSuccess:
+		return "Success: Verification succeeded"
+	case StatusInvalidInclusionProof:
+		return "Invalid inclusion proof detected: Merkle inclusion proof for blob batch is invalid"
+	case StatusSecurityAssumptionsNotMet:
+		return "Security assumptions not met: BLS signer weight is less than the required threshold"
+	case StatusBlobQuorumsNotSubset:
+		return "Blob quorums are not a subset of the confirmed quorums"
+	case StatusRequiredQuorumsNotSubset:
+		return "Required quorums are not a subset of the blob quorums"
+	default:
+		return "Unknown status code"
+	}
+}
+
+type EigenDACert interface {
+	BlobVersion()coreV2.BlobVersion
+	RelayKeys() []coreV2.RelayKey
+	Version() CertificateVersion
+	ReferenceBlockNumber() uint64
+	QuorumNumbers() []byte
+
+	ComputeBlobKey() (*coreV2.BlobKey, error)
+	Commitments() (*encoding.BlobCommitments, error)
+	Serialize() ([]byte, error)
```

Comment:
I think we might need 2 serializations on this interface? One for calling verifyDACert (abiEncode()) and one for serializing for batcher inbox (rlp encode?)

Because right now with this interface change proxy would receive this interface and the only thing it could call is Serialize() which abi encodes, and is probably not what we want to be sending to the batcher inbox?

Reply:
good callout - will add the change

Reply:
[92ab00c](https://github.com/Layr-Labs/eigenda/pull/1532/commits/92ab00cee909375647e0c4177761bd799429dfda)